File: Java/Day15_20_05_2025/src/EncapsulationEmployee.java
/* Q4. Create a class Employee with the following private fields:
name (String)
salary (double)
Implement setter methods with validation rules:
salary should not be negative.
Write a main method to demonstrate these validations. */
class Employee {
    private String name;
    private double salary;
    public void setName(String name) {
        this.name = name; //instance variable
    }
    public String getName() {
        return name;
    }
    public void setSalary(double salary) {
        if (salary >= 0) {
            this.salary = salary;
        } else {
            System.out.println("Invalid salary! Salary cannot be negative.");
        }
    }
    public double getSalary() {
        return salary;
    }
}
public class EncapsulationEmployee {
    public static void main(String[] args) {
        Employee emp = new Employee();
        emp.setName("John Doe");
        emp.setSalary(50000);
        System.out.println("Employee Name: " + emp.getName());
        System.out.println("Salary: $" + emp.getSalary());
        emp.setSalary(-10000);
        System.out.println("Salary after invalid update attempt: $" + emp.getSalary());
    }
}

File: Python/Day21_20_05_2025/EncapsulationEmployee.py
""" Q3. Write a class Employee with a private attribute __salary.
Use the @property decorator to create a getter method.
Use the @<attribute>.setter decorator to set the value of __salary.
Ensure that the salary cannot be set to a negative value. """
class Employee:
    def __init__(self, salary):
        self.__salary = salary
    @property
    def salary(self):
        return self.__salary
    @salary.setter
    def salary(self, value):
        if value < 0:
            print("Error: Salary cannot be negative.")
        else:
            self.__salary = value
emp = Employee(10000)
print("Current Salary:", emp.salary)
emp.salary = 50000
print("Updated Salary:", emp.salary)
emp.salary = -5000